framework/include/utils/Shuffle.h

```diff
@@ -385,6 +419,73 @@ MooseUtils::resample(const std::vector<T> & data,
   return replicate;
 }
 
+template <typename T, typename ActionFunctor>
+void
+MooseUtils::resampleAct(const std::vector<T> & data,
```

Comment:
I like the concept, but don't like the name. It doesn't help me understand what it does, but I can't think of something better. `resampleWithFunctor`?

Reply:
Haha, that was my original name, but I thought it was too long. I will change back.